lib/tests/streamlit/elements/date_input_test.py

```diff
@@ -50,6 +51,16 @@ def test_just_disabled(self):
         c = self.get_delta_from_queue().new_element.date_input
         self.assertEqual(c.disabled, True)
 
+    def test_none_value(self):
+        """Test that it can be called with None as value."""
+        st.date_input("the label", value=None)
+
+        c = self.get_delta_from_queue().new_element.date_input
+        self.assertEqual(c.label, "the label")
+        # If a proto property is null is not determined by this value,
+        # but by the check via the HasField method:
+        self.assertEqual(c.default, [])
```

Comment:
Nit: The others had the `c.HasField` check - is this purposefully missing?

Reply:
Yep, the reason is that date input uses repeatable fields since it also supports range selection, and repeatable fields cannot be made optional in protobuf. Therefore, we are using an empty array to represent the null state of `date_input` which does not support the `HasField` check.

Reply:
I added a comment in the main implementation to explain this aspect